### PR TITLE
CSHARP-4155: Test against MongoDB 6.0

### PR DESCRIPTION
**Pull request by JamesKovacs**

### CODE REVIEW

File: evergreen/evergreen.yml [hunk 1]
```diff
@@ -1330,6 +1330,10 @@ axes:
         display_name: "latest"
         variables:
            VERSION: "latest"
+      - id: "6.0"
```
Comment (DmitryLukyanov) [inline evergreen/evergreen.yml:1333]: we have a number of envs where we run only `5.0` servers like `csfle-with-mocked-kms-tests-windows` or where servers are defined explicitly like in `secure-tests-macOS` or compressions, we may update them too

**Review by DmitryLukyanov — APPROVED** [review]
Question and then LGTM

File: evergreen/evergreen.yml (hunk 1 above)
Comment (JamesKovacs) [reply evergreen/evergreen.yml:1333]: Good catch. I've updated the test matrix.

File: evergreen/evergreen.yml [hunk 2]
```diff
@@ -1466,7 +1470,7 @@ buildvariants:
      - name: test-netstandard21
 
 - matrix_name: "tests-zlib-compression"
-  matrix_spec: { compressor : "zlib", auth: "noauth", ssl: "nossl", version: ["3.6", "4.0", "4.2", "4.4", "5.0", "latest"], topology: "standalone", os: "windows-64" }
+  matrix_spec: { compressor : "zlib", auth: "noauth", ssl: "nossl", version: "*", topology: "standalone", os: "windows-64" }
```
Comment (DmitryLukyanov) [inline evergreen/evergreen.yml:1473]: I've scheduled all variants to check that it works (but it should as I see)

**Pull request merged**